From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 10 Jul 2023 22:05:08 +0200
Subject: [PATCH] Material fixes - Add missing values to hasGravity

Adds the following missing values to `Material.hasGravity`:
* `POINTED_DRIPSTONE`
* `SCAFFOLDING`
* `SUSPICIOUS_GRAVEL` and `SUSPICIOUS_SAND`
Also adds the following missing legacy value:
* `LEGACY_DRAGON_EGG`

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index cb33809b4051276f67955d8b917dd355ff4b60db..56d5e6cb22e8ed4f4cdbbbe5299934fd43a5e63d 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -8878,15 +8878,22 @@ public enum Material implements Keyed, Translatable, net.kyori.adventure.transla
             case MAGENTA_CONCRETE_POWDER:
             case ORANGE_CONCRETE_POWDER:
             case PINK_CONCRETE_POWDER:
+            case POINTED_DRIPSTONE: // Fiddle - add missing Material values with gravity
             case PURPLE_CONCRETE_POWDER:
             case RED_CONCRETE_POWDER:
             case RED_SAND:
             case SAND:
+            // Fiddle start - add missing Material values with gravity
+            case SCAFFOLDING:
+            case SUSPICIOUS_GRAVEL:
+            case SUSPICIOUS_SAND:
+            // Fiddle end - add missing Material values with gravity
             case WHITE_CONCRETE_POWDER:
             case YELLOW_CONCRETE_POWDER:
             // ----- Legacy Separator -----
             case LEGACY_SAND:
             case LEGACY_GRAVEL:
+            case LEGACY_DRAGON_EGG: // Fiddle - add missing Material values with gravity
             case LEGACY_ANVIL:
             case LEGACY_CONCRETE_POWDER:
             //</editor-fold>
